src/subscription_manager/capabilities.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import print_function, division, absolute_import
+
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+
+"""
+Some REST API calls returns still the same result (usually some server capabilities).
+Thus it is wise to cache such result to some file. This modules is used for caching
+such REST API calls.
+"""
+
+import json
+import socket
+import logging
+
+import rhsm.connection as connection
+import subscription_manager.injection as inj
+import subscription_manager.version as version
+
+log = logging.getLogger(__name__)
+
+
+class ServerCache(object):
+    """
+    Class providing cached information about resources and capabilities of candlepin server
+    """
+
+    CACHE_FILE = "/var/lib/rhsm/server_capabilities.json"
+
+    @classmethod
+    def _load_cache_file(cls):
+        """
+        Try to load data from cache file
+        :return:
+        """
+        data = None
+
+        try:
+            with open(cls.CACHE_FILE, 'r') as fp:
+                json_str = fp.read()
+                data = json.loads(json_str)
+        except IOError:
+            pass
+
+        return data
+
+    @classmethod
+    def _write_cache_file(cls, data):
+        """
+        Try to write data to cache file
+        :param data: fresh data
+        :return: None
+        """
+        log.debug("Writing cached server capabilities: %s" % data)
+
+        # Write fresh data to cache file
+        with open(cls.CACHE_FILE, 'w') as fp:
+            json.dump(data, fp)
+
+    @staticmethod
+    def _get_supported_resources_from_server(uep=None):
+        """
+        Try to load supported resources from the server
+        :param uep: connection to candlepin server
+        :return: dictionary of supported resources
+        """
+        resources = {}
+
+        if uep is None:
+            cp_provider = inj.require(inj.CP_PROVIDER)
+            uep = cp_provider.get_consumer_auth_cp()
+        try:
+            resources = uep.get_supported_resources()
+        except (socket.error, connection.ConnectionException) as e:
+            log.exception(e)
+
+        return resources
+
+    @staticmethod
+    def _loaded_data_not_obsolete(data, consumer_uuid):
```

Comment:
Could we switch this definition from "_loaded_data_not_obsolete" to "_loaded_data_obsolete" (and flip the associated return values)? I think this would make reading through it's usage more clear.

Reply:
Sounds good.